bindings/python/src/trainers.rs

```diff
@@ -41,6 +45,34 @@ impl PyTrainer {
         })
     }
 }
+#[pymethods]
+impl PyTrainer {
```

Comment:
The main implementation of pickling.


__getstate__ and __setstate__ are enough to pickle objects in python.